src/pages/index.js

```diff
@@ -22,128 +23,261 @@ import {
 
   profilePopup.open();
 }
 
+function openImageModal(data) {
+  imgagePopup.open(data);
+}
+
 function openAddCardModal() {
   addCardPopup.open();
 }
 
-// Button Event Listeners
+function createCard(data) {
+  const cardElement = new Card(
+    data,
+    cardTemplate,
+    openImageModal,
+    openDeleteModal,
+    handleLikeButton
+  );
+
+  return cardElement.getView();
+}
+
+function openDeleteModal(currentCardId, currentCardElement) {
+  deleteCardPopup.setButtonText("Yes");
+  deleteCardPopup.open();
+  deleteCardPopup.currentCardId = currentCardId;
+  deleteCardPopup.currentCardElement = currentCardElement;
+}
+
+function handleCardDelete() {
+  deleteCardPopup.currentCardElement.remove();
+
+  return api.deleteCard(deleteCardPopup.currentCardId).catch((err) => {
+    console.error("Delete Card Error:", err);
+  });
+}
+
+function handleLikeButton(cardId, isLiked) {
+  console.log(cardId);
+  console.log(isLiked);
+
+  if (!isLiked) {
+    return api
+      .addLike(cardId)
+      .then(() => true)
+      .catch((err) => {
+        console.error("Like Button Error:", err);
+        return false;
+      });
+  } else {
+    return api
+      .removeLike(cardId)
+      .then(() => false)
+      .catch((err) => {
+        console.error("Like Button Error:", err);
+        return true;
+      });
+  }
+}
+
+//// Button Event Listeners ////
+// avatar image click
+profileAvatarWrapper.addEventListener("click", openEditAvatarModal);
+
 // profile edit button click
 profileEditButton.addEventListener("click", openEditProfileModal);
 
 // add card button click
 addCardButton.addEventListener("click", openAddCardModal);
 
-//// Forms ////
-const profileForm = document.forms["edit-profile-form"];
-const addCardForm = document.forms["add-card-form"];
+//// Form Submit Handlers ////
+// Avatar form submission handler
+function handleAvatarFormSubmit(data) {
+  profileAvatar.src = data.link;
 
-//// Form Input Fields ////
-const profileNameInput = profileForm.querySelector(".modal__input_type_name");
-const profileDescriptionInput = profileForm.querySelector(
-  ".modal__input_type_description"
-);
-const newCardTitleInput = addCardForm.querySelector(".modal__input_type_title");
-const newCardUrlInput = addCardForm.querySelector(".modal__input_type_url");
+  return api.updateUserAvatar(data).catch((err) => {
+    console.error(err); // log the error to the console
+  });
+}
 
-//// Form Submit Handlers ////
 // profile form submission handler
 function handleProfileFormSubmit(data) {
-  // insert new values into the textContent property of the
-  // corresponding profile elements
-  user.setUserInfo(data);
-
-  // // close popup
-  // profilePopup.close();
-  // profilePopup.reset();
+  // change propert name of description to about
+  const updatedData = { name: data.name, about: data.description };
+
+  // Update user data on server
+  return api
+    .updateUserInfo(updatedData)
+    .then((res) => {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```